File: app/src/main/java/mydebts/android/app/data/model/Participant.java
package mydebts.android.app.data.model;

import android.support.annotation.NonNull;

public class Participant {
    private Long id;
    private Event event;
    private Person person;
    private Double debt;

    public Participant(Long id, Event event, Person person, Double debt) {
        this.id = id;
        this.event = event;
        this.person = person;
        this.debt = debt;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }

    public Person getPerson() {
        return person;
    }

    public void setPerson(Person person) {
        this.person = person;
    }

    public Double getDebt() {
        return debt;
    }

    public void setDebt(Double debt) {
        this.debt = debt;
    }

    @Override
    public String toString() {
        return "Participant{" +
                "id=" + id +
                ", event=" + event +
                ", person=" + person +
                ", debt=" + debt +
                '}';
    }

    public static Builder builder() {
        return new Builder();
    }

    public static Builder builder(@NonNull Participant source) {
        return builder()
                .id(source.getId())
                .event(source.getEvent())
                .person(source.getPerson())
                .debt(source.getDebt());
    }

    public static class Builder {
        private Long id;
        private Event event;
        private Person person;
        private Double debt;

        public Builder id(Long id) {
            this.id = id;
            return this;
        }

        public Builder event(Event event) {
            this.event = event;
            return this;
        }

        public Builder person(Person person) {
            this.person = person;
            return this;
        }

        public Builder debt(Double debt) {
            this.debt = debt;
            return this;
        }

        public Participant build() {
            return new Participant(id, event, person, debt);
        }
    }
}

File: app/src/main/java/mydebts/android/app/data/model/Participant.kt
package mydebts.android.app.data.model

import android.os.Parcel
import android.os.Parcelable

data class Participant(
        var id: Long? = null,
        var event: Event? = null,
        var person: Person? = null,
        var debt: Double? = null) : Parcelable{
    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Participant> = object : Parcelable.Creator<Participant> {
            override fun createFromParcel(source: Parcel): Participant = Participant(source)
            override fun newArray(size: Int): Array<Participant?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
    source.readValue(Long::class.java.classLoader) as Long?,
    source.readParcelable<Event>(Event::class.java.classLoader),
    source.readParcelable<Person>(Person::class.java.classLoader),
    source.readValue(Double::class.java.classLoader) as Double?
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(id)
        dest.writeParcelable(event, 0)
        dest.writeParcelable(person, 0)
        dest.writeValue(debt)
    }
}
